### PR TITLE
Mark mpTelemetry exposed APIs as third-party

### PR DESCRIPTION
**Pull request by Yas-Franco**
For #23080

### CODE REVIEW

**Comment by Yas-Franco** [comment]
#build

**Comment by Yas-Franco** [comment]
#build

**Comment by Azquelt** [comment]
6e26145 says that's it's updating FAT tests, but it also includes product code changes.

Although the product code changes were needed, the commit message should at least mention the product code changes.

**Comment by Yas-Franco** [comment]
#libby

File: dev/io.openliberty.io.opentelemetry.internal/bnd.bnd [hunk 1]
```diff
@@ -28,26 +28,37 @@ Import-Package: \
...
+  io.opentelemetry.context; version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.trace.export;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.trace.data;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.common;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.autoconfigure.spi.traces;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.autoconfigure.spi;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.resources;version=${openTelemetryVersion},\
+  io.opentelemetry.semconv.trace.attributes;version=${openTelemetryVersion},\
+  io.opentelemetry.semconv.resource.attributes;version=${openTelemetryVersion},\
+  io.opentelemetry.instrumentation.annotations;version=${openTelemetryVersion}
```
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/bnd.bnd]: The versions here are not quite right. We could manually set each one to `version="[${openTelemetryVersion},2)", but if we remove the version from here and add the API bundle to the buildpath then BND should create the correct version range automatically.

File: dev/io.openliberty.io.opentelemetry.internal/bnd.bnd [hunk 2]
```diff
@@ -28,26 +28,37 @@ Import-Package: \
...
+  !io.opentelemetry.context; version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.trace.export;version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.trace.data;version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.common;version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.autoconfigure.spi.traces;version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.autoconfigure.spi;version=${openTelemetryVersion},\
+  !io.opentelemetry.sdk.resources;version=${openTelemetryVersion},\
+  !io.opentelemetry.semconv.trace.attributes;version=${openTelemetryVersion},\
+  !io.opentelemetry.semconv.resource.attributes;version=${openTelemetryVersion},\
+  !io.opentelemetry.instrumentation.annotations;version=${openTelemetryVersion},\
```
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/bnd.bnd]: The version is not needed on these exclusion lines (though I don't think it does any harm)

Comment (tevans78) [reply dev/io.openliberty.io.opentelemetry.internal/bnd.bnd]: best to leave the version off

File: dev/io.openliberty.io.opentelemetry.internal/bnd.bnd (hunk 1 above)
Comment (tevans78) [reply dev/io.openliberty.io.opentelemetry.internal/bnd.bnd]: right, not all of these are going to be version 1.19.0 ... for example some may well be 1.19.0-alpha

File: dev/io.openliberty.mpTelemetry.1.0.thirdparty/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  io.opentelemetry.context; version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.trace.export;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.trace.data;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.common;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.autoconfigure.spi.traces;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.autoconfigure.spi;version=${openTelemetryVersion},\
+  io.opentelemetry.sdk.resources;version=${openTelemetryVersion},\
+  io.opentelemetry.semconv.trace.attributes;version=${openTelemetryVersion},\
+  io.opentelemetry.semconv.resource.attributes;version=${openTelemetryVersion},\
+  io.opentelemetry.instrumentation.annotations;version=${openTelemetryVersion}
```
Comment (tevans78) [inline dev/io.openliberty.mpTelemetry.1.0.thirdparty/bnd.bnd]: some of these versions should not be 1.19.0, some are probably 1.19.0.alpha

File: dev/io.openliberty.io.opentelemetry.internal/bnd.bnd (hunk 1 above)
Comment (yasmin-aumeeruddy) [reply dev/io.openliberty.io.opentelemetry.internal/bnd.bnd:44]: Would it be best to remove the versions from this list then?

**Comment by Yas-Franco** [comment]
#build

**Comment by Yas-Franco** [comment]
#libby

**Review by Azquelt — APPROVED** [review]

**Pull request merged**